bundles/org.openhab.core.io.websocket/src/main/java/org/openhab/core/io/websocket/CommonWebSocketServlet.java

```diff
@@ -94,7 +111,25 @@ private class CommonWebSocketCreator implements WebSocketCreator {
             if (servletUpgradeRequest == null || servletUpgradeResponse == null) {
                 return null;
             }
-            if (isAuthorizedRequest(servletUpgradeRequest)) {
+
+            String accessToken = null;
+            String secWebSocketProtocolHeader = servletUpgradeRequest.getHeader(SEC_WEBSOCKET_PROTOCOL_HEADER);
+            if (secWebSocketProtocolHeader != null) { // if the client sends the Sec-WebSocket-Protocol header
+                // respond with the default protocol
+                servletUpgradeResponse.setHeader(SEC_WEBSOCKET_PROTOCOL_HEADER, WEBSOCKET_PROTOCOL_DEFAULT);
+                // extract the base64 encoded access token from the requested protocols
+                Matcher matcher = WEBSOCKET_ACCESS_TOKEN_PATTERN.matcher(secWebSocketProtocolHeader);
+                if (matcher.find() && matcher.group("base64") != null) {
+                    String base64 = matcher.group("base64");
+                    accessToken = new String(Base64.getDecoder().decode(base64));
```

Comment:
This may throw `IllegalArgumentException` if given token has invalid coding. Should this be handled?

Reply:
Yes, I will fix that.

Reply:
Done